src/net_processing.cpp

```diff
@@ -1441,6 +1443,25 @@ void PeerManagerImpl::AddToCompactExtraTransactions(const CTransactionRef& tx)
     vExtraTxnForCompactIt = (vExtraTxnForCompactIt + 1) % max_extra_txn;
 }
 
+void PeerManagerImpl::MaybeDisconnect(CNode& peer, std::string_view message)
+{
+    if (peer.HasPermission(NetPermissionFlags::NoBan)) {
+        LogPrintLevel(BCLog::NET, BCLog::Level::Warning, "Not disconnecting noban peer %d!\n", peer.GetId());
```

Comment:
Working on the logging in this file right now and saw similar messages; this might be clearer
```suggestion
        LogPrintLevel(BCLog::NET, BCLog::Level::Warning, "not disconnecting misbehaving peer %d because it has noban permission\n", peer.GetId());
```